site/gatsby-node.js

```diff
@@ -1,16 +1,33 @@
 const path = require('path')
+const MiniCssExtractPlugin = require('mini-css-extract-plugin')
+const { Octokit } = require('@octokit/core')
+const octokit = new Octokit()
 
-module.exports.onCreateNode = ({ node, actions, getNode }) => {
+// TODO move all exports to separate files 
+
+module.exports.onCreateNode = async ({ node, actions, getNode }) => {
   if (node.internal.type === 'MarkdownRemark') {
     const fileNode = getNode(node.parent)
     const parsedFilePath = path.parse(fileNode.relativePath)
     const { sourceInstanceName } = fileNode
 
     if (sourceInstanceName === 'protocols') {
-      const protocolName = parsedFilePath.dir
-      const protocolVersion = parsedFilePath.name
+      const [protocolName, protocolVersion] = parsedFilePath.dir.split('/')
       const slug = `${protocolName}/${protocolVersion}/`
 
+      let avatar = ''
+      try {
+        const response = await octokit.request('GET /users/{username}', {
+          username: node.frontmatter.username,
+        })
+        if (response.status >= 400) {
+          throw new Error()
```

Comment:
What would happen if response had code 500? Will it throw exception?

Reply:
I think it should :)